src/controllers/presentation.ts

```diff
@@ -0,0 +1,243 @@
+	 *     parameters:
+	 *       - in: query
+	 *         name: verifyStatus
+	 *         description: If set to `true` the verification will also check the status of the presentation. Requires the VP to have a `credentialStatus` property.
+	 *         schema:
+	 *           type: boolean
+	 *           default: false
+	 *       - in: query
+	 *         name: fetchRemoteContexts
+	 *         description: When dealing with JSON-LD you also MUST provide the proper contexts. * Set this to `true` ONLY if you want the `@context` URLs to be fetched in case they are a custom context.
+	 *         schema:
+	 *           type: boolean
+	 *           default: false
+	 *       - in: query
+	 *         name: allowDeactivatedDid
+	 *         description: If set to `true` allow to verify credential which based on deactivated DID.
+	 *         schema:
+	 *           type: boolean
+	 *           default: false
+	 *     requestBody:
+	 *       content:
+	 *         application/x-www-form-urlencoded:
+	 *           schema:
+	 *             $ref: '#/components/schemas/PresentationVerifyRequest'
+	 *         application/json:
+	 *           schema:
+	 *             $ref: '#/components/schemas/PresentationVerifyRequest'
+	 *     responses:
+	 *       200:
+	 *         description: The request was successful.
+	 *         content:
+	 *           application/json:
+	 *             schema:
+	 *               $ref: '#/components/schemas/IVerifyResult'
+	 *       400:
+	 *         $ref: '#/components/schemas/InvalidRequest'
+	 *       401:
+	 *         $ref: '#/components/schemas/UnauthorizedError'
+	 *       500:
+	 *         $ref: '#/components/schemas/InternalError'
+	 */
+	public async verifyPresentation(request: Request, response: Response) {
+		const result = validationResult(request);
+		if (!result.isEmpty()) {
+			return response.status(StatusCodes.BAD_REQUEST).json({ error: result.array()[0].msg });
+		}
+
+		const { presentation, verifierDid, policies, makeFeePayment } = request.body;
+		const verifyStatus = request.query.verifyStatus === 'true';
+		const allowDeactivatedDid = request.query.allowDeactivatedDid === 'true';
+
+		// define identity service strategy setup
+		const identityServiceStrategySetup = new IdentityServiceStrategySetup(response.locals.customer.customerId);
+		// create cheqd presentation from w3c presentation
+		const cheqdPresentation = new CheqdW3CVerifiablePresentation(presentation);
+		// get holder did
+		const holderDid = cheqdPresentation.holder;
+
+		if (!allowDeactivatedDid) {
+			const resolutionResult = await new IdentityServiceStrategySetup().agent.resolveDid(holderDid);
+
+			if (resolutionResult.didDocumentMetadata.deactivated) {
+				return response.status(StatusCodes.BAD_REQUEST).send({
+					error: `${holderDid} is deactivated`,
+				});
+			}
+		}
+
+		if (makeFeePayment) {
```

Comment:
The condition makes sense, but certainly not by itself.

Following the reasoning stated above, with a `statusList` representing a valid, resolved resource:

```suggestion
		if (makeFeePayment && statusList?.resource?.metadata?.encrypted) {
```